casper/src/main/scala/coop/rchain/casper/api/BlockApiImpl.scala

```diff
@@ -174,11 +179,63 @@ class BlockApiImpl[F[_]: Concurrent: RuntimeManager: BlockDagStorage: BlockStore
       )
   }
 
-  override def deployStatus(deployId: DeployId): F[ApiErr[String]] =
-    BlockDagStorage[F]
-      .asInstanceOf[BlockDagKeyValueStorage[F]]
-      .deployStatus(deployId)
-      .map(_.asRight[Error])
+  override def deployStatus(deployId: DeployId): F[ApiErr[DeployExecStatus]] = {
+
+    def deployBlockHash: F[String] =
+      findDeploy(deployId).map(_.getOrElse(LightBlockInfo.defaultInstance)).map(_.blockHash)
+
+    def getBlockInfo(blockHash: String): F[BlockInfo] =
+      getBlock(blockHash).map(_.getOrElse(BlockInfo.defaultInstance))
```

Comment:
Default block here doesn't make sense. If user receive it in any case it cannot reason abut what happened. If block must  be found raise an error.
Also using BlockApi public methods with already wrapped errors can create another point of confusion. It's wrong that errors are converted to string in BlockApi in the first place. Better to use BlockStore.getUnsafe and BlockDagStorage[F].lookupByDeployId directly.

For these kind of errors user is not responsible so admin of the node should see it in the node log. Ensure that this is the case.